feature/selector.py

```diff
@@ -605,39 +623,84 @@ def _bench(selectors: Dict[str, Union[SelectionMethod.Correlation,
     method_to_runtime = {}
     score_df = pd.DataFrame(index=data.columns)
     selected_df = pd.DataFrame(index=data.columns)
-    for method_name, method in selectors.items():
-        selector = Selective(method)
-        t0 = time()
-        if verbose:
-            print("\n>>> Running", method_name)
-        scores = None
-        selected = []
-        try:
-            subset = selector.fit_transform(data, labels)
-            scores = selector.get_absolute_scores()
-            selected = [1 if c in subset.columns else 0 for c in data.columns]
-            method_to_runtime[method_name] = round((time() - t0) / 60, 2)
-        except Exception as exp:
-            print("Exception", exp)
-            scores = np.repeat(0, len(data.columns))
-            selected = np.repeat(0, len(data.columns))
-            method_to_runtime[method_name] = str(round((time() - t0) / 60, 2)) + " (exception)"
-        finally:
-            score_df[method_name] = scores
-            selected_df[method_name] = selected
+
+    # Find the effective number of jobs
+    size = len(selectors.items())
+    if n_jobs < 0:
+        n_jobs = max(mp.cpu_count() + 1 + n_jobs, 1)
+    n_jobs = min(n_jobs, size)
+
+    # Parallel benchmarks for each method
+    output_list = Parallel(n_jobs=n_jobs, backend=backend, verbose=10, require="sharedmem")(
```

Comment:
Could you say a bit more about why you used require="sharedmem"? I think this will constrain the backend that will actually be used, which might not be obvious to the user. I'm also guessing that it might reduce the memory footprint, but will hurt performance compared to multi-processing.

Reply:
I noticed that without require=“sharedmem”, `Parallel` uses LokyBackend for `n_jobs > 1` which does not improve the runtime. With “sharedmem”, it chooses ThreadingBackend instead which yields better performance. Could we do it a different way in order to avoid the problems with “sharedmem”?

Reply:
If you noticed better performance using "sharedmem" I think we can stick with that.